pandas/core/groupby/groupby.py

```diff
@@ -1120,12 +1117,14 @@ def _resolve_numeric_only(self, numeric_only: bool | lib.NoDefault) -> bool:
             if self.obj.ndim == 2:
                 # i.e. DataFrameGroupBy
                 numeric_only = True
+                obj = self._obj_with_exclusions
```

Comment:
put in a comment in reference to this issue and explain *why* we are doing this (compatiblity)

Reply:
> put in a comment in reference to this issue and explain _why_ we are doing this (compatiblity)

I've done this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712